waku/node/peer_manager/peer_store/waku_peer_storage.nim

```diff
@@ -58,113 +70,104 @@ proc encode*(remotePeerInfo: RemotePeerInfo): PeerStorageResult[ProtoBuffer] =
   for proto in remotePeerInfo.protocols.items:
     pb.write(3, proto)
 
-  try:
-    pb.write(4, remotePeerInfo.publicKey)
-  except ResultError[CryptoError] as e:
-    return err("Failed to encode public key")
+  let catchRes = catch: pb.write(4, remotePeerInfo.publicKey)
+  if catchRes.isErr():
+    return err("Enncoding public key failed: " & catchRes.error.msg)
+
+  pb.write(5, uint32(ord(remotePeerInfo.connectedness)))
+
+  pb.write(6, uint64(remotePeerInfo.disconnectTime))
+
+  if remotePeerInfo.enr.isSome():
+    pb.write(7, remotePeerInfo.enr.get().raw)
 
-  ok(pb)
+  return ok(pb)
 
 ##########################
 # Storage implementation #
 ##########################
 
 proc new*(T: type WakuPeerStorage, db: SqliteDatabase): PeerStorageResult[T] =
-  ## Misconfiguration can lead to nil DB
+  # Misconfiguration can lead to nil DB
   if db.isNil():
     return err("db not initialized")
 
-  ## Create the "Peer" table
-  ## It contains:
-  ##  - peer id as primary key, stored as a blob
-  ##  - stored info (serialised protobuf), stored as a blob
-  ##  - last known enumerated connectedness state, stored as an integer
-  ##  - disconnect time in epoch seconds, if applicable
-
-  # TODO: connectedness and disconnectTime are now stored in the storedInfo type
-  let
-    createStmt = db.prepareStmt("""
-      CREATE TABLE IF NOT EXISTS Peer (
-          peerId BLOB PRIMARY KEY,
-          storedInfo BLOB,
-          connectedness INTEGER,
-          disconnectTime INTEGER
-      ) WITHOUT ROWID;
-      """, NoParams, void).expect("this is a valid statement")
-
-  let res = createStmt.exec(())
-  if res.isErr:
+  # Create the "Peer" table
+  # It contains:
+  #  - peer id as primary key, stored as a blob
+  #  - stored info (serialised protobuf), stored as a blob
+  let createStmt = db.prepareStmt(
+    """
+    CREATE TABLE IF NOT EXISTS Peer (
+        peerId BLOB PRIMARY KEY,
```

Comment:
Maybe not needed but if we are creating a different schema, maybe it is worth moving the previous data to the current one. Nevertheless, I don't consider this very critical, though.

Reply:
@Ivansete-status In your opinion, is peer persistence enabled often when running nodes?

Reply:
Hey @SionoiS ! Sorry for the late response. I have the feeling that peer persistence is enabled by default and therefore, used extensively.